### ISSUE TITLE
[QUESTION] Using callbacks (early stopping, logging, etc)

### ISSUE THREAD

**Issue by FMelloMascarenhas-Cohere** [huggingface/setfit#308]
Hi all, thanks for your work here!

**TLDR**: Is there a way to add callbacks for early stopping and logging (for example, with W&B?).

I am using setfit for a project, but I could not figure out a way to add early stopping. I am afraid that I am overfitting to the training set. I also cant really say that I am, because I am not sure how I can log the training metrics (train/eval performance across epochs).

I saw that the script [run_full.py](https://github.com/huggingface/setfit/blob/ebee18ceaecb4414482e0a6b92c97f3f99309d56/scripts/transformers/run_full.py#L104) has it, but I couldn't figure out how to do it with SetFit API.

thanks!

**Comment by tomaarsen** [comment]
Hello!

The `SetFitTrainer` instance does not support callbacks at this time. We're planning to support that after #265 is merged for a v1.0.0 release. The problem is that the trainer differs from the Hugging Face Transformer Trainer, and so it doesn't support all of its functionality.

Alternatively, these `fit` calls would need to be updated to include the `callback`.
https://github.com/huggingface/setfit/blob/8c8b8f0e235ddb23a2bb061d3afa0824f6a735c3/src/setfit/trainer.py#L373-L394

- Tom Aarsen

**Comment by FMelloMascarenhas-Cohere** [comment]
Thanks! I unfortunately don't have time to help with it at this moment, but I am looking forward for new releases.
